Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/handlers.go

```diff
@@ -413,6 +415,65 @@ func (r *requestAttributeGetter) GetAttribs(req *http.Request) authorizer.Attrib
 	return &attribs
 }
 
+func WithActingAs(handler http.Handler, requestContextMapper api.RequestContextMapper, a authorizer.Authorizer) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		requestedSubject := req.Header.Get("Impersonate-User")
+		if len(requestedSubject) == 0 {
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		ctx, exists := requestContextMapper.Get(req)
+		if !exists {
+			forbidden(w, req)
+			return
+		}
+		requestor, exists := api.UserFrom(ctx)
+		if !exists {
+			forbidden(w, req)
+			return
+		}
+
+		actingAsAttributes := &authorizer.AttributesRecord{
+			User:     requestor,
+			Verb:     "impersonate",
+			APIGroup: api.GroupName,
+			Resource: "users",
+			// ResourceName:    requestedSubject,
```

Comment:
> This is commented-out here with no explanation. Should this be removed or explained?

Needs to be updated upstream now that other pulls have merged, I'll open the upstream for it.  That code is dead in origin, so I'm disinclined to pick a second pull unless you feel strongly.